src/interfaceCallForwarder.ts

```
@@ -0,0 +1,395 @@
+      node.kind,
+      node.abstract,
+      node.fullyImplemented,
+      node.linearizedBaseContracts,
+      node.usedErrors,
+      node.documentation,
+      node.children,
+      node.nameLocation,
+      node.raw,
+    );
+    result.push(writer.write(solContract).slice(0, -1));
+    result.push('\n    // Write your logic here\n');
+    result.push('}\n');
+    return result;
+  }
+}
+
+class CornerTypeNameWriter extends ASTNodeWriter {
+  writeInner(node: CornerTypeName, _writer: ASTWriter): SrcDesc {
+    const result: SrcDesc = [];
+    result.push(node.typeString);
+    return result;
+  }
+}
+
+class CornerStructTypeNameWriter extends ASTNodeWriter {
+  writeInner(node: CornerTypeName, _writer: ASTWriter): SrcDesc {
+    const result: SrcDesc = [];
+    result.push(node.typeString);
+    return result;
+  }
+}
+
+class CornerTypeName extends TypeName {
+  constructor(id: number, src: string, typeString: string) {
+    super(id, src, typeString);
+  }
+}
+
+class CornerStructTypeName extends TypeName {
+  constructor(id: number, src: string, typeString: string) {
+    super(id, src, typeString);
+  }
+}
+
+function getSolTypeName(cairoTypeString: string, ast: AST, struct_names: string[]): TypeName {
+  if (cairoTypeString.endsWith('*')) {
+    const baseTypeName: TypeName = getSolTypeName(cairoTypeString.slice(0, -1), ast, struct_names);
+    return createArrayTypeName(baseTypeName, ast);
+  }
+  if (cairoTypeString === 'felt') {
+    return createUintNTypeName(248, ast);
+  }
+  if (cairoTypeString === 'Uint256') {
+    return createUint256TypeName(ast);
+  }
+  if (struct_names.includes(cairoTypeString)) {
+    // TODO: replace this with a proper struct type
+    // after creation of struct dependency ordering
+    return new CornerStructTypeName(ast.reserveId(), '', `<STRUCT:${cairoTypeString}>`);
+  }
+  // throw new Error(`Cairo type: ${cairoTypeString} Not Supported`);
+  return new CornerTypeName(ast.reserveId(), '', `<TYPE:${cairoTypeString}>`);
```

We should never hit this case and if we do we should throw an error